test/e2e/auth/subjectreviews.go

```diff
@@ -0,0 +1,156 @@
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+var _ = SIGDescribe("SubjectReview", func() {
+	f := framework.NewDefaultFramework("subjectreview")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	ginkgo.It("should support SubjectReview API operations", func() {
+
+		AuthClient := f.ClientSet.AuthorizationV1()
+		ns := f.Namespace.Name
+		saName := "sa-" + utilrand.String(5)
+
+		ginkgo.By(fmt.Sprintf("Creating a Serviceaccount %q in namespace %q", saName, ns))
+
+		sa, err := f.ClientSet.CoreV1().ServiceAccounts(ns).Create(context.TODO(), &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: saName,
+			},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Unable to create serviceaccount %q", saName)
+
+		saUsername := serviceaccount.MakeUsername(sa.Namespace, sa.Name)
+		framework.Logf("saUsername: %q", saUsername)
+		saGroups := []string{"system:authenticated"}
+		saGroups = append(saGroups, serviceaccount.MakeGroupNames(sa.Namespace)...)
+		framework.Logf("saGroups: %#v", saGroups)
+		saUID := string(sa.UID)
+		framework.Logf("saUID: %q", saUID)
+
+		ginkgo.By(fmt.Sprintf("Creating clientset to impersonate %q", saUsername))
+		config := rest.CopyConfig(f.ClientConfig())
+		config.Impersonate = rest.ImpersonationConfig{
+			UserName: saUsername,
+			UID:      saUID,
+			Groups:   saGroups,
+		}
+
+		impersonatedClientSet, err := kubernetes.NewForConfig(config)
+		framework.ExpectNoError(err, "Could not load config, %v", err)
+
+		ginkgo.By(fmt.Sprintf("Creating SubjectAccessReview for %q", saUsername))
+
+		sar := &authorizationv1.SubjectAccessReview{
+			Spec: authorizationv1.SubjectAccessReviewSpec{
+				ResourceAttributes: &authorizationv1.ResourceAttributes{
+					Verb:      "list",
+					Resource:  "configmaps",
+					Namespace: ns,
+					Version:   "v1",
+				},
+				User:   saUsername,
+				Groups: saGroups,
+				UID:    saUID,
+			},
+		}
+
+		sarResponse, err := AuthClient.SubjectAccessReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Unable to create a SubjectAccessReview, %#v", err)
+		framework.Logf("sarResponse Status: %#v", sarResponse.Status)
+		expectedAllowed := sarResponse.Status.Allowed
+
+		ginkgo.By(fmt.Sprintf("Verifying as %q api 'list' configmaps in %q namespace", saUsername, ns))
+		_, err = impersonatedClientSet.CoreV1().ConfigMaps(ns).List(context.TODO(), metav1.ListOptions{})
```

Comment:
I'd suggest saving the actual request error for later use instead of shadowing:

```suggestion
		_, requestErr := impersonatedClientSet.CoreV1().ConfigMaps(ns).List(context.TODO(), metav1.ListOptions{})
```